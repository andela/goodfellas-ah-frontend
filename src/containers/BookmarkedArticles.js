import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userPlaceholderImage } from '../mixin';
import { Loader as Loading } from '../components/shared/Loading';
import { fetchProfile } from '../actions/profileActions';
import { getBookmarkedArticles } from '../actions/articleActions';
import BookmarkedArticleList from '../components/article/BookmarkedArticleList';

// eslint-disable-next-line react/prefer-stateless-function
class BookmarkedArticles extends Component {
  componentWillMount = () => {
    const {
      user: { userId },
      fetchProfile: fetchUserProfile,
      getBookmarkedArticles: fetchBookmarkedArticles,
    } = this.props;
    fetchUserProfile(userId);
    fetchBookmarkedArticles();
  }

  render() {
    const { profileStore, bookmarked } = this.props;
    const fullName = `${profileStore.user.firstname} ${profileStore.user.lastname}`;
    if (profileStore.loading && !bookmarked) return <Loading />;
    return (
      <div>
        <div className="profile-details">
          <img alt="profile" className="profile-image" src={profileStore.profile.image || userPlaceholderImage} />
          <h3 id="user-name" className="username">{fullName}</h3>
        </div>
        <h3 className="profile-title">Bookmarks</h3>
        <BookmarkedArticleList articles={bookmarked} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  profileStore: state.profile,
  user: state.auth.ownProfile,
  bookmarked: state.articles.bookmarkedArticles,
});

export default connect(mapStateToProps, { fetchProfile, getBookmarkedArticles })(BookmarkedArticles);
