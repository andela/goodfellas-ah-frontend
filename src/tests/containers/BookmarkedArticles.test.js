import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import BookmarkedArticles from '../../containers/BookmarkedArticles';
import Root from '../../root';

let wrapped;

const mockState = {
  auth: {
    authenticated: null,
    userId: null,
    user: null,
    ownProfile: { userId: 1 },
  },
};

beforeEach(() => {
  wrapped = mount(
    <Root initialState={mockState}>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <BookmarkedArticles />
      </MemoryRouter>
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('BookmarkedArticles UI', () => {
  it('view should render as expected', () => {
    const tree = toJson(wrapped);
    expect(tree).toMatchSnapshot();
  });
});
