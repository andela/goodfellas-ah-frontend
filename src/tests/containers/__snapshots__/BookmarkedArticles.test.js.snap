// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkedArticles UI view should render as expected 1`] = `
<_default
  initialState={
    Object {
      "auth": Object {
        "authenticated": null,
        "ownProfile": Object {
          "userId": 1,
        },
        "user": null,
        "userId": null,
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(BookmarkedArticles)>
          <BookmarkedArticles
            fetchProfile={[Function]}
            getBookmarkedArticles={[Function]}
            profileStore={
              Object {
                "articles": Array [],
                "favorites": Array [],
                "followers": Array [],
                "following": Array [],
                "loading": true,
                "profile": Object {},
                "profileError": "",
                "profileView": "Following",
                "user": Object {},
              }
            }
            user={
              Object {
                "userId": 1,
              }
            }
          >
            <Loader>
              <div
                className="no-record centralizer"
              >
                <div>
                  <img
                    alt="loader"
                    src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/Spinner-1s.gif"
                  />
                    
                  <span>
                    Loading
                  </span>
                </div>
              </div>
            </Loader>
          </BookmarkedArticles>
        </Connect(BookmarkedArticles)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
